android/sdl_android/src/main/java/com/smartdevicelink/transport/SdlBroadcastReceiver.java

```diff
@@ -145,6 +147,15 @@ public void onReceive(Context context, Intent intent) {
             DebugTool.logError(TAG, "You cannot use the default SdlRouterService class, it must be extended in your project. THIS WILL THROW AN EXCEPTION IN FUTURE RELEASES!!");
         }
 
+        Hashtable<String, Object> vehicleInfoStore = (Hashtable<String, Object>) intent.getSerializableExtra(TransportConstants.CONNECT_VEHICLE_INFO);
+
+        VehicleType vehicleType;
```

Comment:
@kostyaBoss I would move `VehicleType vehicleType` from line 152 and put it on line 92 (top level variable).

Reply:
the thing is, moving forward and looking into the code of the wakeUpRouterService, we will see, that the param passed should be final. If we declare it as final at the top-level we will not be able to update it eventually